component/file_cache/file_cache.go

```diff
@@ -691,69 +691,88 @@ func (fc *FileCache) DeleteFile(options internal.DeleteFileOptions) error {
 	return nil
 }
 
-// OpenFile: Makes the file available in the local cache for further file operations.
-func (fc *FileCache) OpenFile(options internal.OpenFileOptions) (*handlemap.Handle, error) {
-	log.Trace("FileCache::OpenFile : name=%s, flags=%d, mode=%s", options.Name, options.Flags, options.Mode)
+func (fc *FileCache) DownloadFile(options internal.DownloadFileOptions) (*handlemap.Handle, error) {
+
+	//extract flag and mode out of the value from handle
+	var flag int
+	var fMode fs.FileMode
+
+	flagMode, found := options.Handle.GetValue("fileFlagMode")
```

Comment:
`flagMode` and `flagModeStruct` have confusingly similar names. Since the first one is really just a temporary variable for untyped data, you could use the pattern we have in other places where `GetValue` is called, where the first untyped throw-away variable is just called `val`, and then the meaningful variable name is saved for when the type is asserted.